MoreLinq.Test/DuplicatesTest.cs

```diff
@@ -0,0 +1,140 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2023 Julien Aspirot. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class DuplicatesTest
+    {
+        [Test]
+        public void Duplicates_IsLazy()
+        {
+            _ = new BreakingSequence<object>().Duplicates();
+        }
+
+        [Test]
+        public void Streams_Duplicates_As_They_Are_Discovered()
+        {
+            static IEnumerable<string> Source()
+            {
+                yield return "DUPLICATED_STRING";
+                yield return "DUPLICATED_STRING";
+                throw new TestException();
+            }
+
+            using var source = TestingSequence.Of(Source());
+
+            void Act() => source.Duplicates().Take(1);
+
+            Assert.That(Act, Throws.Nothing);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_Without_Duplicates_Then_Empty_Sequence_Is_Returned()
+        {
+            using var testingSequence = TestingSequence.Of("FirstElement", "SecondElement", "ThirdElement");
+
+            var duplicates = testingSequence.Duplicates();
```

Comment:
To be consistent with all other tests, compose with `ToArray()` here too:

```suggestion
            var duplicates = testingSequence.Duplicates().ToArray();
```

Reply:
done